packages/core/_src/stm/TMap/operations/delete.ts

```diff
@@ -0,0 +1,41 @@
+import { concreteTArray } from "@effect/core/stm/TArray/operations/_internal/InternalTArray"
+import { TMap } from "@effect/core/stm/TMap/definition"
+import { concreteTMap } from "@effect/core/stm/TMap/operations/_internal/InternalTMap"
+
+/**
+ * Removes binding for given key.
+ *
+ * @tsplus fluent ets/TMap delete
+ */
+export function delete_<K, V>(self: TMap<K, V>, k: K): USTM<void> {
+  concreteTMap(self)
+  return STM.Effect((journal) => {
+    const buckets = self.tBuckets.unsafeGet(journal)
+
+    concreteTArray(buckets)
+
+    const idx = TMap.indexOf(k, buckets.chunk.length)
+    const bucket = buckets.chunk.unsafeGet(idx)!.unsafeGet(journal)
+
+    const partitions = bucket.partition((_) => _.get(0) === k)
```

Comment:
Would use `Equals.equals` for the equality check here since Scala's `==` doesn't technically perform reference equality.